vscode-swift-language-server/src/swiftSourceTypes.ts

```diff
@@ -1,306 +1,291 @@
-    Swift |
-    Decl |
-    DeclExtension |
-    DeclFunction |
-    DeclFunctionOperator |
-    DeclFunctionAccessor |
-    DeclFunctionMethod |
-    DeclVar |
-    Ref |
-    RefFunction |
-    RefFunctionMethod |
-    RefFunctionOperator |
-    RefVar |
-    Syntaxtype |
-    SyntaxtypeAttribute;
+    let Keyword: 'source.lang.swift.keyword';
+    let DeclAssociatedtype: 'source.lang.swift.decl.associatedtype';
+    let DeclClass: 'source.lang.swift.decl.class';
+    let DeclEnum: 'source.lang.swift.decl.enum';
+    let DeclEnumelement: 'source.lang.swift.decl.enumelement';
+    let DeclGenericTypeParam: 'source.lang.swift.decl.generic_type_param';
+    let DeclProtocol: 'source.lang.swift.decl.protocol';
 
-/**
- * Maps SourceKit types to VSCode {SymbolKind}
- * FIXME: kinda limted
- *
- * @export
- * @param {SwiftType} swiftType
- * @returns {SymbolKind}
- */
-export function symbolKindForSwiftType(swiftType: SwiftType): SymbolKind {
-    // not really easy to have an obvious completion kind per type.
-    switch (swiftType) {
-// high confidence
-        case Decl.Module:
-            return SymbolKind.Module;
-        case Decl.Enum:
-        case Decl.Enumelement:
-        case Ref.Enum:
-        case Ref.Enumelement:
-            return SymbolKind.Enum;
-// medium confidence
-        case DeclFunction.Free:
-        case DeclFunction.Subscript:
-        case DeclFunctionOperator.Infix:
-        case RefFunction.Free:
-        case RefFunction.Subscript:
-        case RefFunctionOperator.Infix:
-            return SymbolKind.Function;
-        case DeclFunctionMethod.Class:
-        case DeclFunctionMethod.Static:
-        case DeclFunctionMethod.Instance:
-            return SymbolKind.Method;
-        case DeclFunction.Constructor:
-        case RefFunction.Constructor:
-            return SymbolKind.Constructor;
-        case DeclFunctionAccessor.Getter:
-        case DeclFunctionAccessor.Setter:
-            return SymbolKind.Field;
-        case Decl.Associatedtype:
-        case Decl.Class:
-        case Decl.Struct:
-        case DeclExtension.Class:
-        case Ref.Class:
-        case Ref.Struct:
-            return SymbolKind.Class;
-        case RefFunctionMethod.Class:
-        case RefFunctionMethod.Instance:
-            return SymbolKind.Method;
-// low confidence
-        case DeclVar.Global:
-        case DeclVar.Instance:
-        case DeclVar.Local:
-        case RefVar.Global:
-        case RefVar.Instance:
-        case RefVar.Local:
-            return SymbolKind.Variable;
-        case Syntaxtype.Argument:
-        case Syntaxtype.Parameter:
-            return SymbolKind.Field;
-        case Ref.Associatedtype:
-        case Decl.GenericTypeParam:
-        case Ref.GenericTypeParam:
-        case Ref.Protocol:
-        case Ref.Typealias:
-        case Decl.Protocol:
-        case Decl.Typealias:
-            return SymbolKind.Interface;
+    let DeclStruct: 'source.lang.swift.decl.struct';
```

Comment:
I'm also a typescript novice so some conventions might not be idiomatic typescript
From what I read typescript doesn't easily allow something like
```ts
enum Foo {
  bar = "baz"
}
let foo = Foo.bar;
switch foo {
case bar: 
  console.log("a bar");
}
```
We can try a union type, but that looks like 
```ts
type Foo = 'baz' | 'bar'
let foo: Foo = 'baz';
switch foo {
case 'baz': 
  console.log("a baz")
}
```
That leaves us stuck with `string` like thing everywhere still. A long way to say I'm sure there's a better way to do this that I don't know how to. That way might still use String literal types.